.circleci/config.yml

```diff
@@ -1,24 +1,125 @@
 version: 2.1
 
+# add your orb below, to be used in integration tests (note: a @dev:alpha
+# release must exist; you'll need to publish manually at least once)
+# you can use scripts/publish-alpha.sh to publish a @dev:alpha release
 orbs:
-  orb-tools: circleci/orb-tools@8.27.3
-  shellcheck: circleci/shellcheck@1.3.14
-
-jobs:
-  getting-started:
-    docker:
-      - image: circleci/node
-    steps:
-      - run:
-          name: Getting Started
-          command: |
-            echo "Your orb is now building on CircleCI"
-            echo "This is a temporary configuration file. Continue with the orb-init.sh script and follow the README.md to publish your orb."
+  build-orb: netgaintechnology/build-orb@volatile
+  prebuild-orb: netgaintechnology/prebuild-orb@volatile
+  orb-tools: circleci/orb-tools@volatile
+
 workflows:
-  setup:
+  # this `lint-pack_validate_publish-dev` workflow will run on any commit
+  lint-pack_validate_publish-dev:
     jobs:
-      - getting-started
-      - orb-tools/lint
-      - shellcheck/check:
-          ignore: 'SC1090,SC2034,SC2059,SC2001,SC2143,SC2046'
+      # lint your destructured orb YAML and markdown files
+      - prebuild-orb/yaml_lint
+      - prebuild-orb/markdown_lint
+
+      # pack your orb YAML files to a single orb.yml
+      # validate the orb.yml file to ensure it is well-formed
+      - orb-tools/pack:
+          requires:
+            - prebuild-orb/yaml_lint
+
+      # release dev version of orb, for testing & possible publishing
+      # requires a CircleCI API token to be stored as CIRCLE_TOKEN (default)
+      # https://circleci.com/docs/2.0/managing-api-tokens
+      # store CIRCLE_TOKEN as a project env var or Contexts resource
+      # if using Contexts, add your context below
+      - orb-tools/publish-dev:
+          orb-name: netgaintechnology/build-orb
+          requires:
+            - orb-tools/pack
+
+      # trigger an integration workflow to test the dev version of your orb
```

Comment:
If it's just doing checkouts, we can use a github oauth2 token.  I keep one in secrets manager already, and it just means we use the https versus ssh urls for github.  The main nicety is that terraform can use the github token, and we don't store two creds for the same service.